### PR TITLE
Fixing DNS hostname code to save info to DB in taskmanager.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Got rid of update_hostname method.
- Return the entry.name in the create_instance_entry method, and now use that value to update the DB in the taskmanager code.
- Fixed hostname bug in the views.

### CODE REVIEW

File: reddwarf/instance/models.py [hunk 1]
```diff
@@ -366,8 +365,6 @@ def create(cls, context, name, flavor_id, image_id,
         service_status = InstanceServiceStatus.create(instance_id=db_info.id,
             status=ServiceStatuses.NEW)
 
-        dns_client = create_dns_client(context)
-        dns_client.update_hostname(db_info)
```
Comment (hub-cap) [inline reddwarf/instance/models.py:370]: will removing this cause the create call to _not_ return the hostname? it should be in there

Comment (TimSimpsonR) [inline reddwarf/instance/models.py:370]: After create you won't see it, no. It will appear once it's ready or not appear at all should something go wrong.

Comment (hub-cap) [inline reddwarf/instance/models.py:370]: I don't think it's an option. It's contract breaking to _not_ show on create. :(

Comment (TimSimpsonR) [inline reddwarf/instance/models.py:370]: Darn, you're right. Switched it back to save it originally.

Comment (hub-cap) [inline reddwarf/instance/models.py:370]: heart

**Comment by hub-cap** [comment]
im pretty sure the `create_instance_entry` will still gen a hostname and save it instead of using the one we saved originally

**Comment by TimSimpsonR** [comment]
I changed my copy to save the hostname as some silly string in reddwarf_api, and ran a create instance. I saw the hostname prov in the task manager logs, but when I checked the database finally the old silly name remained.  :confounded:

**Comment by hub-cap** [comment]
+1

**Comment by ed-** [comment]
+1

**Comment by cp16net** [comment]
+1

**Comment by rnirmal** [comment]
+1

**Pull request merged**